src/version-util.ts

```diff
@@ -54,44 +53,13 @@ export async function getLatestGcloudSDKVersion(): Promise<string> {
  *
  */
 async function getGcloudVersion(url: string): Promise<string> {
-  const u = new URL(url);
-
-  const opts = {
-    hostname: u.hostname,
-    port: u.port,
-    path: u.pathname + u.search,
-    method: 'GET',
-    headers: {
-      'User-Agent': userAgentString,
-    },
-  };
-
-  const resp: string = await new Promise((resolve, reject) => {
-    const req = https.request(opts, (res) => {
-      res.setEncoding('utf8');
-
-      let body = '';
-      res.on('data', (data) => {
-        body += data;
-      });
-
-      res.on('end', () => {
-        if (res.statusCode && res.statusCode >= 400) {
-          reject(body);
-        } else {
-          resolve(body);
-        }
-      });
-    });
-
-    req.on('error', (err) => {
-      reject(err);
-    });
-
-    req.end();
-  });
+  const http = new httpm.HttpClient(userAgentString);
+  const res = await http.get(url);
+  if (res.message.statusCode && res.message.statusCode != 200) {
+    throw new Error(`Failed to retrieve gcloud SDK version, statusCode: ${res.message.statusCode}`);
+  }
 
-  const body = JSON.parse(resp);
+  const body = JSON.parse(await res.readBody());
   const version = body.version;
   if (!version) {
     throw new Error(`Failed to retrieve gcloud SDK version, invalid response body: ${body}`);
```

Comment:
I think this is actually a bug. We should pull the `await res.readBody()` into its own call so we can actually print the response body.

Reply:
Given the recent gcloud issue, I'm hyper-sensitive to swallowing output 😦. Here's what I did in secretmanager: https://github.com/google-github-actions/get-secretmanager-secrets/pull/200

Reply:
Makes sense, I switched to that approach.